recipe-server/client/control/components/Fields.js

```diff
@@ -50,34 +50,43 @@ buildControlField.propTypes = {
   children: pt.node,
 };
 
+export const CheckboxGroup = ({ name, onChange, options = [], value }) => {
+  /**
+   * Checkbox change event handler. Appends or removes the selected checkbox's
+   * value to the existing `value` prop, and reports the change up to redux-form.
+   *
+   * @param  {Event} onChange event object
+   */
+  const handleChange = ({ event: target }) => {
```

Comment:
Can you elaborate on this? `handleChange` interacts with the `value` prop passed into CheckboxGroup, and I'm unsure of how nesting subcomponents would avoid rerenders as a result.

For instance:

```
options.map((option, index) =>
  <CheckboxGroupField
    key={index}
    option={option}
    onChange={onChange}
    checked={value.includes(option.value)}
  />
)
```
Passing redux-form's `onChange` directly (as shown here) means that we would need to pass the `value` into the CheckboxGroupField as well, which would cause a rerender. Using `onChange={handleChange}`, where `handleChange` is defined in CheckboxGroup, would loop back around to the issue that `handleChange` is defined every render, which re-renders the subcomponents.

Furthermore, since these 'components' are not actual React Components, I'm fairly positive that the whole list is rendered regardless of what prop changes are passed in. Perhaps we should actually just use React.Component and define `handleChange` as a component function, which would avoid all of this.